src/sas/qtgui/Perspectives/Fitting/ConstraintWidget.py

```diff
@@ -434,37 +434,107 @@ def onTabCellEdit(self, row, column):
-            param = item.data(0)[item.data(0).index(':') + 1:item.data(0).index('=')].strip()
-            model = item.data(0)[:item.data(0).index(':')].strip()
-            tab = self.available_tabs[model]
-            if isinstance(tab, FittingWidget):
-            # Update the fitting widget whenever a constraint is activated/deactivated
-                if item.checkState() == QtCore.Qt.Checked:
-                    font = QtGui.QFont()
-                    font.setItalic(True)
-                    brush = QtGui.QBrush(QtGui.QColor('blue'))
-                    tab.modifyViewOnRow(tab.getRowFromName(param), font=font, brush=brush)
-                else:
-                    tab.modifyViewOnRow(tab.getRowFromName(param))
-                return
-        # Update the constraint formula
+        # extract information from the constraint object
         constraint = self.available_constraints[row]
-        function = item.text()
-        # remove anything left of '=' to get the constraint
-        function = function[function.index('=')+1:]
-        # No check on function here - trust the user (R)
-        if function != constraint.func:
-            # This becomes rather difficult to validate now.
-            # Turn off validation for Edit Constraint
-            constraint.func = function
-            constraint.validate = False
+        model = constraint.value_ex[:constraint.value_ex.index(".")]
+        param = constraint.param
+        function = constraint.func
+        tab = self.available_tabs[model]
+        # Extract information from the text in the table
+        constraint_text = item.data(0)
+        # Basic sanity check of the string
+        # First check if we have an acceptable sign
+        if "=" not in constraint_text:
+            msg = ("Incorrect operator in constraint definition. Please use = "
+                   "sign to define constraints.")
+            QtWidgets.QMessageBox.critical(
+                self,
+                "Inconsistent constraint",
+                msg,
+                QtWidgets.QMessageBox.Ok)
+            self.initializeFitList()
+            return
+        # Then check if the parameter is correctly defined with colons
+        # separating model and parameter name
+        parameter_full_name = constraint_text[:constraint_text.index(
+            "=")].strip()
+        if ":" not in parameter_full_name:
+            msg = ("Incorrect constrained parameter definition. Please use "
+                   "colons to separate model and parameter on the rhs of the "
+                   "definition, e.g. M1:scale")
+            QtWidgets.QMessageBox.critical(
+                self,
+                "Inconsistent constraint",
+                msg,
+                QtWidgets.QMessageBox.Ok)
+            self.initializeFitList()
+            return
+        # We can parse the string
+        new_param = item.data(0)[
+                item.data(0).index(':') + 1:item.data(0).index('=')].strip()
```

Comment:
PEP8 asks for spaces around the `:` if it is using complex expressions.

Or use a pair of splits:
```
item.data(0).split("=", 1)[0].split(":", 1)[1].strip()
```
or maybe a regular expression:
```
re.search(r":(\w*) *=", item.data(0))[1]
```